modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotTraverse.scala

```diff
@@ -29,17 +23,25 @@ object GlobalSnapshotTraverse {
   def make[F[_]: Sync](
     loadInc: Hash => F[Option[Signed[GlobalIncrementalSnapshot]]],
     loadFull: Hash => F[Option[Signed[GlobalSnapshot]]],
+    loadInfo: SnapshotOrdinal => F[Option[GlobalSnapshotInfo]],
     contextFns: SnapshotContextFunctions[F, GlobalSnapshotArtifact, GlobalSnapshotContext],
     rollbackHash: Hash
   ): GlobalSnapshotTraverse[F] =
     new GlobalSnapshotTraverse[F] {
       val logger = Slf4jLogger.getLogger[F]
 
       def loadChain(): F[(GlobalSnapshotInfo, Signed[GlobalIncrementalSnapshot])] = {
-        def loadFullOrErr(h: Hash) =
-          loadFull(h).flatMap(_.liftTo[F](new Throwable(s"Global snapshot not found during rollback, hash=${h.show}")))
         def loadIncOrErr(h: Hash) =
-          loadInc(h).flatMap(_.liftTo[F](new Throwable(s"Incremental snapshot not found during rollback, hash=${h.show}")))
+          loadInc(h).flatMap(_.liftTo[F](new Exception(s"Incremental snapshot not found during rollback, hash=${h.show}")))
+        def loadInfoOrErr(o: SnapshotOrdinal) =
+          loadInfo(o).flatMap(_.liftTo[F](new Exception(s"Expected SnapshotInfo not found during rollback, ordinal=${o.show}")))
```

Comment:
Well, it depends on case, because sometimes you have more than one ordinal that you show (eg. actual vs expected). We've used this pattern `name=${value}` across the codebase more than 15 times for just ordinal (not counting other properties), so I'd keep using it unless we decide to change it in all places.